tools/c7n_tencentcloud/tests/test_tc_tags.py

```diff
@@ -0,0 +1,299 @@
+            for tag in resource["Tags"]:
+                if tag["Key"] == "tag_add_test_key_for_test" and \
+                        tag["Value"] == "tag_add_test_value_for_test":
+                    tag_add_success = True
+                    break
+            assert tag_add_success
+
+    @pytest.mark.vcr
+    def test_modify_tag(self, options, ctx):
+        policy = self.load_policy(
+            {
+                "name": "cvm-test-rename-tag",
+                "resource": "tencentcloud.cvm",
+                "query": [{
+                    "InstanceIds": ["ins-00lycyy6", "ins-nhhm5ppo"]
+                }],
+                "actions": [
+                    {
+                        "type": "rename-tag",
+                        "old_key": "tag_add_test_key_for_test",
+                        "new_key": "tag_add_test_key_for_test_rename"
+                    }
+                ]
+            },
+            config=options
+        )
+        resources = policy.run()
+        assert len(resources) == 2
+        for resource in resources:
+            tag_exist = False
+            for tag in resource["Tags"]:
+                if tag["Key"] == "tag_add_test_key_for_test":
+                    tag_exist = True
+                    break
+            assert tag_exist
+        if self.recording:
+            time.sleep(10)
+        policy = {
+            "name": "cvm-test",
+            "query": [{
+                "InstanceIds": ["ins-00lycyy6", "ins-nhhm5ppo"]
+            }],
+        }
+        cvm = CVM(ctx, policy)
+        resources = cvm.resources()
+        for resource in resources:
+            old_key_not_exist = True
+            new_key_exist = False
+            for tag in resource["Tags"]:
+                if tag["Key"] == "tag_add_test_key_for_test":
+                    old_key_not_exist = False
+                    break
+                if tag["Key"] == "tag_add_test_key_for_test_rename":
+                    new_key_exist = True
+            assert old_key_not_exist and new_key_exist
+
+    @pytest.mark.vcr
+    def test_remove_tag(self, options, cvm):
+        policy = self.load_policy(
+            {
+                "name": "cvm-test-remove-tag",
+                "resource": "tencentcloud.cvm",
+                "query": [{
+                    "InstanceIds": ["ins-00lycyy6"]
+                }],
+                "actions": [
+                    {
+                        "type": "remove-tag",
+                        "tags": ["tag_add_test_key_for_test_rename"]
+                    }
+                ]
+            },
+            config=options
+        )
+        resources = policy.run()
+        tag_exist = False
+        for tag in resources[0]["Tags"]:
+            if tag["Key"] == "tag_add_test_key_for_test_rename":
+                tag_exist = True
+                break
+        assert tag_exist
+        if self.recording:
+            time.sleep(10)
+        resources = cvm.resources()
```

Comment:
this fetches all the instances instead of just the one that's being queried. typically we would do this by just using a client class to fetch the tags against the qcs to verify which avoids the extraneous api call to refetch the resource state.